py/strato/common/log/log2text.py

```diff
@@ -227,20 +227,79 @@ def printLogs(logFiles, formatter):
 
 
 def updateConfFile(field, value):
-    with open(LOG_CONFIG_FILE_PATH, 'r') as f:
-        conf = yaml.load(f.read())
+    try:
+        with open(LOG_CONFIG_FILE_PATH, 'r') as f:
+            conf = yaml.load(f.read())
+    except:
+        print "No configuration file was found. creating new"
+        conf = {}
+    finally:
         conf[field] = value
     with open('/tmp/stratolog-conf.tmp', 'w') as f:
         f.write(yaml.dump(conf))
+        f.close()
     os.system('sudo mv /tmp/stratolog-conf.tmp %s' % LOG_CONFIG_FILE_PATH)
 
 
-def universalOpen(filePath, mode='r'):
+def universalOpen(filePath, mode='r', tail=0):
     if filePath.endswith('.gz'):
-        return gzip.GzipFile(filePath, mode)
+        fileObj = gzip.GzipFile(filePath, mode)
     else:
-        return open(filePath, mode)
-
+        fileObj = open(filePath, mode)
+    if tail > 0:
+        tailFile(fileObj, tail)
+    return fileObj
+
+def tailFile(fileObj, n):
+    bufferSize = bufferIncrement = 8192
+    fileObj.seek(0,2)
+    fileSize = fileObj.tell()
+    while bufferSize < fileSize:
+        fileObj.seek(-bufferSize,2)
+        linesInBuffer = len([line for line in fileObj.readlines() if line.strip()])
+        if linesInBuffer > n:
+            fileObj.seek(-bufferSize,2)
+            while linesInBuffer != n:
+                fileObj.readline()
+                linesInBuffer -= 1
+            return fileObj
+        else:
+            bufferSize += bufferIncrement
+    fileObj.seek(0, 0)
+    return fileObj
+
+def ping(host):
+    return os.system("ping -c 1 %s > /dev/null 2>&1" % host) == 0
+
+def runRemotely(host, ignoreArgs):
+    try:
+        configFile = yaml.load(open(LOG_CONFIG_FILE_PATH, 'r').read())
+    except:
+        configFile = {}
+        print "Configuration file was not found"
+
+    args = [arg for arg in sys.argv[1:] if arg not in ignoreArgs]
+    command = "strato-log %s" % ' '.join(args)
+
+    sshCommand = 'sshpass -p %(password)s ssh -o ServerAliveInterval=5 -o ServerAliveCountMax=1 -o ' \
+                 'StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null %(hostname)s %(command)s | less -r'
+    hostname = "%(user)s@%(hostname)s" % dict(user=configFile.get("defaultRemoteUser", 'root'),
+                                              hostname=findHostname(host))
```

Comment:
Please make all the resolution before you generate the string command - it wuold be much easier to follow.